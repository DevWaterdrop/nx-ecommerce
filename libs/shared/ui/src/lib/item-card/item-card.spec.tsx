import { render } from '@testing-library/react';

import ItemCard from './item-card';

describe('ItemCard', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<ItemCard />);
    expect(baseElement).toBeTruthy();
  });
});
