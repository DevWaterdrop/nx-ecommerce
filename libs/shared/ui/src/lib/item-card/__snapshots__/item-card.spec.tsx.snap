// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard should render successfully 1`] = `
<div
  className="relative flex flex-col list-none w-60"
  data-testid="item-card"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <a
    className="mb-1"
    href="/i/lovely-flower"
    onClick={[Function]}
    onMouseEnter={[Function]}
    title="Lovely flower, For you, for me, for us, price: 65"
  >
    <div
      className="relative w-full mb-2 object-cover bg-gray-100 h-72"
    >
      Next image stub
      <div
        aria-hidden="true"
        className="absolute left-0 top-0 h-full w-full transition-opacity opacity-0"
        data-testid="second-image"
      >
        Next image stub
      </div>
    </div>
    <div
      className="flex flex-col gap-1"
    >
      <span
        className="text-sm font-semibold uppercase sm:(text-base)"
      >
        Lovely flower
      </span>
      <span
        className="text-xs sm:(text-sm max-w-3/5)"
      >
        For you, for me, for us
      </span>
      <span
        className="font-bold text-md sm:(text-xl)"
      >
        65.00$
      </span>
    </div>
  </a>
  <div
    className="flex gap-2 md:(absolute right-0 bottom-0) "
  >
    <div
      className="relative sm:(opacity-0)"
    >
      <button
        className="inline-block cursor-pointer transform-gpu transition-all active:scale-92 rounded-full p-3 text-white bg-sky-600 hover:(bg-sky-700) active:(bg-sky-800) flex items-center"
        onClick={[Function]}
        title="add to cart"
        type="button"
      >
        <div
          aria-hidden="true"
          className="relative flex items-center select-none pointer-events-none h-4 w-4  sm:(h-6 w-6)"
        >
          Next image stub
        </div>
      </button>
    </div>
    <button
      className="inline-block cursor-pointer transform-gpu transition-all active:scale-92 rounded-full p-3 hover:(bg-gray-100) active:(bg-gray-300) flex items-center sm:(opacity-0)"
      onClick={[Function]}
      title="add to favorite"
      type="button"
    >
      <div
        aria-hidden="true"
        className="relative flex items-center select-none pointer-events-none h-4 w-4  sm:(h-6 w-6)"
      >
        Next image stub
      </div>
    </button>
  </div>
</div>
`;
